### PR TITLE
Add support for ECS metadata labels.

### PR DESCRIPTION
**Pull request by easwars**

### CODE REVIEW

**Comment by ejona86** [comment]
Can we please not mix the bug fix (server_features) with a new feature (metadata)?

**Comment by easwars** [comment]
> Can we please not mix the bug fix (server_features) with a new feature (metadata)?

Was taking some liberties since you made me a contributor on this repo :P 
Reverted the other changes now.

**Comment by ejona86** [comment]
Sanjay is going to discuss with some others to determine the requirements for this.

**Comment by ejona86** [comment]
The verdict from Sanjay's meeting is we don't need to manage these as a single string. Let's do the "pass the flag multiple times" approach and get rid of the comma separated stuff. I think we'll also want to give it a more generic name like `--metadata` or the like, and not assume it is tied to ECS.

**Comment by easwars** [comment]
> The verdict from Sanjay's meeting is we don't need to manage these as a single string. Let's do the "pass the flag multiple times" approach and get rid of the comma separated stuff. I think we'll also want to give it a more generic name like `--metadata` or the like, and not assume it is tied to ECS.

Thanks. Done. PTAL.

File: main.go [hunk 1]
```diff
@@ -73,6 +77,7 @@ func main() {
 		zone:               zone,
 		includeV3Features:  *includeV3Features,
 		includePSMSecurity: *includePSMSecurity,
+		ecsMetadataLabels:  nodeMetadata,
```
Comment (ejona86) [inline main.go:80]: Remove the "ecs" references.

File: main.go [hunk 2]
```diff
@@ -234,6 +243,24 @@ func getFromMetadata(urlStr string) ([]byte, error) {
 	return body, nil
 }
 
+// Parse the comma separated list of labels specified on the command line as
+// part of the --ecs-metadata-labels flag into a map of key value pairs.
+func parseMetadataLabels(labels string) (map[string]string, error) {
```
Comment (ejona86) [inline main.go:248]: Delete?

File: map_flag.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			b.WriteRune(',')
+		}
+		b.WriteString(key)
+		b.WriteRune('=')
+		b.WriteString((*s)[key])
+	}
+	return b.String()
+}
+
+func (s *stringMapVal) Get() interface{} {
```
Comment (ejona86) [inline map_flag.go:59]: What uses this?

File: main.go (hunk 1 above)
Comment (easwars) [reply main.go]: Done.

File: main.go (hunk 2 above)
Comment (easwars) [reply main.go]: Done. Thanks.

File: map_flag.go (hunk 3 above)
Comment (easwars) [reply map_flag.go]: Looks like I can get rid of it. The `flag` package says that every `Value` type exported by it implements this `Getter` interface. But I think it is mostly for use with `flag.Lookup()`. I don't think we would need that. So, got rid of it.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**